oak_attestation_verification/src/verifier.rs

```diff
@@ -465,6 +462,9 @@ fn verify_root_layer(
     endorsements: Option<&RootLayerEndorsements>,
     reference_values: &RootLayerReferenceValues,
 ) -> anyhow::Result<()> {
+    if let Some(insecure_values) = reference_values.insecure.as_ref() {
+        return verify_insecure(insecure_values).context("insecure root layer verification failed");
+    }
     match values.report.as_ref() {
```

Comment:
The rest here still reads a bit backwards though. I think the main switch should be on the reference values, not the actual value.